framework/src/st_things/things_stack/memory/things_string.c

```diff
@@ -0,0 +1,41 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "things_string.h"
+
+#include <string.h>
+#include <assert.h>
+#include "things_malloc.h"
+
+#define TAG "THINGS_STRING"
+
+char *things_strdup(const char *str)
+{
+    if (!str) {
+        return NULL;
+    }
+
+    // Allocate memory for original string length and 1 extra byte for '\0'
+    size_t length = strlen(str);
+    char *dup = (char *)things_malloc(length + 1);
+    if (NULL != dup) {
+        memcpy(dup, str, length + 1);
+    }
+
```

Comment:
Dont you think copying actual data of length size and you appending '\0'  at the end is more readable and conveys the purpose